### ISSUE TITLE
Does not work with latest version of flow

### ISSUE THREAD

**Issue by Alxandr** [andreypopp/validated#29]
I get 

```
Missing type annotation for `$NonMaybeType`.
```

from the `AnyNode` class. I'm not sure how best to fix this :-/.

**Comment by LoganBarnett** [comment]
Looks like there's some problems with `$NonMaybeType` and polymorphic types: https://github.com/facebook/flow/issues/2236

**Comment by Alxandr** [comment]
You linked to a year old issue for something that just started happening?

**Comment by tmowad** [comment]
I see what Alxandr sees.  More context added in case it helps.  This is with flow-bin 0.71.0.

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/validated/lib/schema.js.flow:158:45

Missing type annotation for $NonMaybeType.

     155│   }
     156│ }
     157│
     158│ export class AnyNode<V: mixed> extends Node<$NonMaybeType<V>> {
     159│   validate(context: Context): ValidateResult<$NonMaybeType<V>> {
     160│     return context.unwrap(value => {
     161│       if (value == null) {



Found 1 error
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by LoganBarnett** [comment]
@Alxandr I don't know why this is new for this library, but yes the issue I linked is indeed still open. It has something to do with a recent version of Flow. Downgrading has been our approach (we're on 0.64 and everything works presently). We tried the upgrade and got this failure. I did some research and the linked ticket is the closest thing I could find to it. Perhaps it's a regression?

FWIW I didn't make the connection that @andreypopp had been participating on the ticket until after I'd linked it. I'd like to keep my participation here productive - If there's a better way for me to signal "I'm having this issue too, and this is what I was able to find relating to it", please let me know. Thanks!

**Comment by Alxandr** [comment]
The type itself does not produce any error though. For instance, the following works:

```js
// @flow

function test<T: mixed>(v: $NonMaybeType<T>): $NonMaybeType<T> {
  return v;
}

class Foo<T> {}

class Bar<T: mixed> extends Foo<$NonMaybeType<T>> {}
```

([tryflow](https://flow.org/try/#0PTAEAEDMBsHsHcBQjIFcB2BjALgS1uqNgKYDO2APACoBcoAtrgB7EAmAfABQBudAJADkCAWQCGATwBGxKuIAOxauwCU-IejFSZ8xVXagA3olCgATsWypThbgG5EAX2SZoo0qVAAxWLCWGniC5uHgBCoqbUdIwsHKDETCTorB7evoIiEtKyCkr6Bg5AA))

**Comment by forticulous** [comment]
In case anyone is interested, it looks like this error occurs in 0.70 and up, 0.69 does not have this issue

**Comment by pronebird** [comment]
Closed by #30. Can others confirm? @LoganBarnett
